realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -134,8 +135,11 @@ public boolean contains(Object object) {
         boolean contains = false;
         if (object instanceof RealmObjectProxy) {
             RealmObjectProxy proxy = (RealmObjectProxy) object;
-            Row row = proxy.realmGet$proxyState().getRow$realm();
-            contains = !(row instanceof InvalidRow) && collection.contains((UncheckedRow) row);
+            // TODO: Maybe we should just let OS throw?
```

Comment:
Yes, the goal right now is upgrading without breaking any existing functionality. We can revisit it for 3.0